File: LojadeCarro/src/main/java/com/javacar/lojadecarro/dto/response/VeiculoDetalheResponse.java
package com.javacar.lojadecarro.dto.response;

import com.javacar.lojadecarro.enums.StatusVeiculo;
import io.swagger.v3.oas.annotations.media.Schema;

import java.math.BigDecimal;
import java.time.LocalDateTime;
import java.util.List;

@Schema(description = "Detalhes de um veículo anunciado")
public record VeiculoDetalheResponse(
        Long id,
        String marca,
        String modelo,
        String carroceria,
        String cor,
        String combustivel,
        String motor,
        BigDecimal valor,
        Integer quilometragem,
        Short anoFabricacao,
        StatusVeiculo statusVeiculo,
        String descricao,
        LocalDateTime dataCadastro,
        VendedorResumoResponse vendedor,
        List<VeiculoImagemResponse> imagens,
        List<OpcionalResponse> opcionais
) {
}

File: LojaDeCarroWeb/src/app/features/veiculos/models/veiculo-detalhe-response.ts
import { StatusVeiculo } from './veiculo-response';

export interface VeiculoImagemResponse {
  id: number;
  principal: boolean;
}

export interface VendedorResumoResponse {
  id: number;
  nome: string;
}

export interface OpcionalResponse {
  id: number;
  nome: string;
  ativo: boolean;
}

export interface VeiculoDetalheResponse {
  id: number;
  marca: string;
  modelo: string;
  carroceria: string;
  cor: string;
  combustivel: string;
  motor: string;
  valor: number;
  quilometragem: number;
  anoFabricacao: number;
  statusVeiculo: StatusVeiculo;
  descricao: string | null;
  dataCadastro: string;
  vendedor: VendedorResumoResponse;
  imagens: VeiculoImagemResponse[];
  opcionais: OpcionalResponse[];
}